### DISCUSSION TITLE
Help using MineStomPVP

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1176] [Help] [answered]
So I'm using https://github.com/TogAr2/MinestomPvP however I cant figure out how to actually make the player take damage.

My code is
```java
GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
globalEventHandler.addListener(PlayerLoginEvent.class, event -> {
    final Player player = event.getPlayer();
    event.setSpawningInstance(instanceContainer);
    player.setRespawnPoint(new Pos(-51, 68, -225));
    player.setGameMode(GameMode.SURVIVAL);
    PvpExtension.setLegacyAttack(player, true);
});
// Init PVP Extension
PvpExtension.init();
MinecraftServer.getGlobalEventHandler().addChild(PvpExtension.events());
```

However, whenever I take fall damage or receive damage from another player it always does 0 damage

**Comment by TogAr2** [comment]
Can you check if the events are actually being called? You can add a breakpoint to `AttackManager#entityHit` and see if it reaches that point

Also make sure you're not cancelling any event related to pvp (both from Minestom and from MinestomPvP)

Edit: do you mean there is no damage being applied, or is the player flashing red but the damage is 0?

**Reply by para-dise** [reply]
The player is flashing red but the damage is 0

**Reply by TogAr2** [reply]
You are using `PvpExtension.events()` instead of `legacyEvents()`, thats' probably the issue

**Reply by para-dise** [reply]
Hmm, so is there no way of having legacy damage without calculating it myself? Or do newer clients just not support the legacy events

(nevermind I see now)

**Comment by TogAr2** [comment, marked as answer]
You are using `PvpExtension.events()` instead of `legacyEvents()`.
If you want modern pvp, remove `PvpExtension.setLegacyAttack(player, true);`

**Reply by para-dise** [reply]
Aside from this, the issue was also that I had the jar in the extensions folder while also loading it in the code
